### PR TITLE
feat!: remove ServiceMonitor CRD

### PR DESCRIPTION
**Pull request by ckaenzig**
## Description of the changes

Remove ServiceMonitor CRD. It causes conflict issues with the CRD installed by the kube-prometheus-stack chart when the verisons are not in sync. It's a bad solution to include the CRD in this module.

This was found because the latest kube-prometheus-chart adds a `attachMetadata` parameter in the ServiceMonitor spec, and the non up-to-date CRD in this module causes Argo CD to keep the resource out-of-sync indefinitely.

Also add a module variable `enable_service_monitor` that's true by default, and allows installing the chart with the ServiceMonitor resources in case there is a dependency issue with the CRD.

## Breaking change

- [x] Yes (in our custom resources in the templates directory of this module)

## Tests executed on which distribution(s)

- [ ] KinD
- [ ] AKS (Azure)
- [ ] EKS (AWS)
- [ ] Scaleway
- [ ] SKS (Exoscale)

### CODE REVIEW

**Comment by ckaenzig** [comment]
I added a variable `enable_service_monitor` to address the missing CRD issue when necessary. As discussed, we could add a similar variable to other modules that deploy such resources.

**Comment by ckaenzig** [comment]
... aaand I rebased on main to cleanup the commits in this branch.

File: variables.tf [hunk 1]
```diff
@@ -45,6 +45,12 @@ variable "custom_solver_configurations" {
   default     = []
 }
 
+variable "enable_service_monitor" {
+  description = "Enable prometheus ServiceMonitor in helm chart."
```
Comment (lentidas) [inline variables.tf:49]:
```suggestion
  description = "Enable Prometheus ServiceMonitor in the Helm chart."
```

**Review by lentidas — APPROVED** [review]
Thank you, @ckaenzig, that was exactly what I was thinking about. I proposed a reword in the comment. Other than that everything looks good.

**Review by jbarascut — APPROVED** [review]

**Review by lentidas — CHANGES_REQUESTED** [review]
Sorry, one important thing I forgot before!

We need to add a way to propagate the variable from the variants to the root module when instantiating cert-manager from a variant. For example, in the [`eks/main.tf`](./eks/main.tf):

module "cert-manager" {
  source = "../"

  cluster_name     = var.cluster_name
  base_domain      = var.base_domain
  argocd_namespace = var.argocd_namespace

  namespace              = var.namespace
  target_revision        = var.target_revision
  deep_merge_append_list = var.deep_merge_append_list
  app_autosync           = var.app_autosync
  enable_service_monitor = var.enable_service_monitor # THIS LINE

  helm_values = concat(local.helm_values, var.helm_values)

  dependency_ids = var.dependency_ids
}
```

**Comment by ckaenzig** [comment]
> We need to add a way to propagate the variable from the variants to the root module when instantiating cert-manager from a variant

Good catch, thanks! I added this as well as your suggested change for the variable description.

**Pull request merged**